jaxopt/_src/zoom_linesearch.py

```diff
@@ -633,9 +636,9 @@ def init_state(
       _, aux = self._fun_with_aux(params, *args, **kwargs)
 
     if descent_direction is None:
-      descent_direction = tree_scalar_mul(-1.0, grad)
+      descent_direction = tree_scalar_mul(-1.0, tree_conj(grad))
 
-    slope = tree_vdot(grad, descent_direction)
+    slope = tree_vdot(grad, descent_direction).real
```

Comment:
Same as for lbfgs: 
Just to be sure that I understand:
- in theory the dot product should be real because the function is real and one rounds it for numerical stability or does the algorithm really require taking the real part?

I ask because in e.g. the implementation of linear solvers in complex values similar issues arose and different patches for numerical stability where used. 
Once we have a test on a simple function we may select what's the best if the problem is in terms of numerical stability

Reply:
...same as above